### PR TITLE
Bugfix/sankey download 434

### LINKED ISSUE #434: [BUG]: Sankey diagram downloads as blank image
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

I installed the app locally, logged in with my gmail address, and got to the dashboard with my emails. I clicked on the button to download the sankey diagram, but the .png file that was downloaded was a blank (white) image.

Image

### Expected Behavior

I expected a sankey diagram of my emails to be downloaded.

### Steps To Reproduce

1. Log in with Google
2. Load dashboard
3. Click button to download sankey diagram
4. Open downloaded file

### Environment

```markdown
- OS: macOS Ventura 13.0.1
- Node: v18.15.0
- npm: 9.5.0
```

### Criteria for Completion

- [ ] When a user goes to download a sankey diagram from their dashboard, the downloaded file should show a diagram instead of a blank image.

### Anything else?

_No response_

### Code of Conduct

- [x] I've read the Code of Conduct and understand my responsibilities as a member of the jobba.help Community

### PR DESCRIPTION
**Pull request by ChrysKoum**
# Pull Request

## Description
Replaces strict, exact-string matching in `process_sankey()` with flexible, keyword-based matching to categorize LLM-generated email statuses. Prevents blank/white PNG downloads by ensuring every status falls into one of the Sankey chart buckets or a fallback “No Response” group.

**Related Issue**  
Fixes #434

---

## Type of Change  
_Select all that apply:_  
- [x] 🐛 Bug fix (non-breaking change which fixes an issue)  
- [ ] ✨ New feature (non-breaking change which adds functionality)  
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] 📝 Documentation update  
- [ ] 🎨 Style update (formatting, renaming)  
- [ ] ♻️ Code refactor (no functional changes)  
- [ ] ⚡️ Performance improvement  
- [x] ✅ Test update  
- [ ] 🔨 Build configuration update  
- [ ] 🔒 Security update  

---

## Changes Made
- Switched from `status == "offer"`-style checks to `any(keyword in status for keyword in …)` for all categories (offer, rejected, availability, interview, no response).  
- Added broad keyword lists to capture natural-language variants (e.g. “offer made,” “congratulations,” “pleased to offer,” “when are you available,” “no reply,” etc.).  
- Introduced fallback logic: if none of the keywords match, classify under **No Response**.  
- Bumped Sankey chart fonts (title: 22 px, labels: 18 px) and increased node padding for readability.  
- Added unit tests covering all new keyword categories and edge-case statuses to prevent regressions.

---

## Screenshots

| Before (blank PNG) | After (rendered Sankey) |
|:------------------:|:-----------------------:|
| sankey_diagram_2025-07-12_before| sankey_diagram_2025-07-12_after  |

---

### Steps to Reproduce
1. Log in via Google.  
2. Navigate to the dashboard.  
3. Click **Download Sankey Diagram**.  
   - **Previous behavior:** PNG is blank/white.  
   - **New behavior:** PNG shows full job-application flow.

### Test Environment
- **OS:** Windows 10  
- **Python:** 3.11+  
- **Backend:** FastAPI + Plotly (Kaleido)  
- **Test Runner:** pytest  

---

## Checklist
- [x] My code follows the style guidelines of this project  
- [x] I performed a self-review of my own code  
- [x] I commented my code, particularly in hard-to-understand areas  
- [ ] I made corresponding changes to the documentation  
- [x] My changes generate no new warnings  
- [x] I added tests that prove my fix is effective  
- [x] New and existing unit tests pass locally  
- [x] Any dependent changes have been merged and published downstream  
- [x] I have checked my code and corrected any misspellings  

---

## Dependencies
_No new production dependencies._  
_Test dependencies:_  
- `pytest`

---

## Database Changes
- [x] No database changes  
- [ ] Database changes included (describe below)

---

## API Changes
- [x] No API changes  
- [ ] API changes included (describe below)

---

## Performance Impact
- [x] No measurable performance impact  
- [ ] Performance impact (describe below)

> The keyword checks add negligible CPU overhead but eliminate the need for repeated blank-image retries.

---

## Security Considerations
- [x] No security implications  
- [ ] Security implications considered (describe below)

---

## Additional Notes
**Root Cause:** Exact-match logic did not capture LLM’s natural-language status variations, leaving all flows uncategorized.  
**Solution Overview:** Replaced exact equals with `any(k in status for k in keywords)` for each category, plus a final fallback.

---

## Reviewer Notes
- The change is localized to `process_sankey()`—easy to audit.  
- Test suite now covers every keyword category and fallback path.  
- Backward-compatible: existing endpoints and chart output unchanged for already-matched statuses.

---

## Post-Deployment Steps
- [x] No post-deployment steps needed  
- [ ] Post-deployment steps required (describe below)

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the project’s license._

### CODE REVIEW

**Comment by ChrysKoum** [comment]
I will take a look and make path commit

**Comment by ChrysKoum** [comment]
@lnovitz Fixed everything waiting for reveiw

**Comment by lnovitz** [comment]
Can you also please add an image in the PR displaying the fallback diagram and what that looks like? 🙏

File: backend/routes/file_routes.py [hunk 1]
```diff
@@ -111,32 +111,74 @@ async def process_sankey(request: Request, db_session: database.DBSession, user_
...
-        elif status == "rejected":
+        elif any(keyword in status for keyword in ["reject", "rejected", "rejection", "regret", "unfortunately"]):
             num_rejected += 1
-        elif status == "request for availability":
+        elif any(keyword in status for keyword in ["availability", "request for availability", "schedule", "when are you available"]):
             num_request_for_availability += 1
-        elif status == "interview scheduled":
+        elif any(keyword in status for keyword in ["interview", "call", "meeting", "invite"]):
+            num_interview_scheduled += 1
+        elif any(keyword in status for keyword in ["no response", "no reply", "unresponsive", "freeze", "hold", "paused", "canceled"]):
```
[the remark this answers is not in the document]
Comment (ChrysKoum) [reply backend/routes/file_routes.py:129]: Yes, I just saw it and will update accordingly. I was using these statuses based on an old version of llm_utils, and I also added variations in case the LLM made mistakes and didn't write the exact status.

File: backend/routes/file_routes.py [hunk 2]
```diff
@@ -101,42 +101,185 @@ async def process_sankey(request: Request, db_session: database.DBSession, user_
...
+    node_labels = [label for label in fig.data[0].node.label if label is not None]
+    node_colors = fig.data[0].node.color[:len(node_labels)]
+    
+    fig.data[0].node.label = node_labels
+    fig.data[0].node.color = node_colors
+    
+    # Add comprehensive layout configuration
+    fig.update_layout(
+        title={
+            'text': f"Job Application Flow Analysis - Status Breakdown<br><sub>Total Applications: {num_applications} | Showing 11 LLM Categories + Other (False Positives Excluded)</sub>",
```
Comment (lnovitz) [inline backend/routes/file_routes.py:271]:
```suggestion
            'text': f"Job Application Flow Analysis - Status Breakdown<br><sub>Total Applications: {num_applications})</sub>",
```
I like that you share the total count here

**Review by lnovitz — APPROVED** [review]
